src/main/java/com/puppycrawl/tools/checkstyle/checks/ArrayTypeStyleCheck.java

```diff
@@ -61,17 +68,18 @@
     @Override
     public void visitToken(DetailAST ast) {
         final DetailAST typeAST = ast.getParent();
-        if (typeAST.getType() == TokenTypes.TYPE
-                // Do not check method's return type.
-                // We have no alternatives here.
-                && typeAST.getParent().getType() != TokenTypes.METHOD_DEF) {
+        if (typeAST.getType() == TokenTypes.TYPE) {
             final DetailAST variableAST = typeAST.getNextSibling();
             if (variableAST != null) {
-                final boolean isJavaStyle =
-                    variableAST.getLineNo() > ast.getLineNo()
+                final boolean isMethod = typeAST.getParent().getType() == TokenTypes.METHOD_DEF;
+                final boolean isJavaStyle = variableAST.getLineNo() > ast.getLineNo()
                     || variableAST.getColumnNo() - ast.getColumnNo() > -1;
 
-                if (isJavaStyle != javaStyle) {
+                // force all methods to be Java style (C functions can't return arrays)
```

Comment:
> C functions can't return arrays

Wording sounds weird. Functions can return arrays, they just can't be written with the array declaration on the method name.